### PR TITLE
Vite build migration

### PR DESCRIPTION
**Pull request by shakiba**
Key changes:
- Build using vite (much faster and simpler), remove direct Rollup build
- Manually add constructor factory types, remove ts transformer

Other changes:
- Remove unused packages
- Update typedoc packages, and update api docs
- ~Add unminified umd build~
- ~Drop sourcemap for unminified build files~

Notes:
- This will be released as v1.1.*
- Constructor factory types will be dropped from v2

### CODE REVIEW

**Review by zOadT — COMMENTED** [review]
Looks great! 🚀 

Two small additional things I noticed:

* "Drop sourcemap for unminified build files": I think there is actually still value in source maps for unminified files as we are still compiling from TS to JS (in es5) and additionally bundle the files into one. Personally I think I would prefer to still get errors regarding the TS files

* From my understanding planck.js is now CJS and I think before it was UMD? Maybe that could lead to some confusion especially since the planck.min.js file is UMD. I think we can just keep planck.js UMD, I don't know if there really is much value in a pure CJS file nowadays?

File: tsconfig.json [hunk 1]
```diff
@@ -1,26 +1,22 @@
...
+    "strictNullChecks": false,
+    "noImplicitAny": false,
     "lib": ["ES2015", "DOM"],
     "stripInternal": true,
     "target": "ES5",
     "esModuleInterop": true,
     "moduleResolution": "Node",
     "resolveJsonModule": true,
+    "allowJs": true,
+    "typeRoots": ["./types"]
```
[the remark this answers is not in the document]
Comment (shakiba) [reply tsconfig.json]: Build fails without this

**Comment by shakiba** [comment]
@zOadT Replaced cjs with umd, and added back sourcemap.

**Comment by shakiba** [comment]
It seems linter doesn't work now, since we updated the typescript version.

Need to migrate eslint in another PR, which requires migrating to new config file format.

**Pull request merged**